### PR TITLE
OSDOCS#12418: KubeVirt CSI Security and Isolation for hosted clusters

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OSDOCS-12418

Link to docs preview: https://85569--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-manage/hcp-manage-virt.html#hcp-virt-storage-security-isolation_hcp-manage-virt

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/hcp-virt-storage-security-isolation.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="hcp-virt-storage-security-isolation_{context}"]
+= KubeVirt CSI storage security and isolation
+
+KubeVirt Container Storage Interface (CSI) extends storage capabilities of the underlying infrastructure cluster to hosted clusters. The CSI driver ensures secure and isolated access to the infrastructure storage classes and hosted clusters by using the following security constraints:
+
+* The storage of a hosted cluster is isolated from the other hosted clusters.
+
+* Worker nodes in a hosted cluster does not have a direct API access to the infrastructure cluster. The hosted cluster can provision storage on the infrastructure cluster only through the controlled KubeVirt CSI interface.
```
Comment (davidvossel) [inline modules/hcp-virt-storage-security-isolation.adoc:13]: should the word `does` change to `do`?

**Review by davidvossel — COMMENTED** [review]
looks good. i made one grammar comment

File: modules/hcp-virt-storage-security-isolation.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-manage/hcp-manage-virt.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="hcp-virt-storage-security-isolation_{context}"]
+= KubeVirt CSI storage security and isolation
+
+KubeVirt Container Storage Interface (CSI) extends storage capabilities of the underlying infrastructure cluster to hosted clusters. The CSI driver ensures secure and isolated access to the infrastructure storage classes and hosted clusters by using the following security constraints:
```
Comment (lahinson) [inline modules/hcp-virt-storage-security-isolation.adoc:9]:
```suggestion
KubeVirt Container Storage Interface (CSI) extends the storage capabilities of the underlying infrastructure cluster to hosted clusters. The CSI driver ensures secure and isolated access to the infrastructure storage classes and hosted clusters by using the following security constraints:
```
On this line and elsewhere this file, should "infrastructure cluster" be "management cluster"?

File: modules/hcp-virt-storage-security-isolation.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+* The storage of a hosted cluster is isolated from the other hosted clusters.
+
+* Worker nodes in a hosted cluster do not have a direct API access to the infrastructure cluster. The hosted cluster can provision storage on the infrastructure cluster only through the controlled KubeVirt CSI interface.
+
+* The hosted cluster cannot access arbitrary storage volumes on the infrastructure cluster that are not associated with the hosted cluster. Because, the hosted cluster does not have access to the KubeVirt CSI cluster controller. The KubeVirt CSI cluster controller runs in a pod in the hosted control plane namespace.
```
Comment (lahinson) [inline modules/hcp-virt-storage-security-isolation.adoc:15]:
```suggestion
* The hosted cluster does not have access to the KubeVirt CSI cluster controller. As a result, the hosted cluster cannot access arbitrary storage volumes on the infrastructure cluster that are not associated with the hosted cluster. The KubeVirt CSI cluster controller runs in a pod in the hosted control plane namespace.
```

File: modules/hcp-virt-storage-security-isolation.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+* Role-based access control (RBAC) of the KubeVirt CSI cluster controller limits the persistent volume claim (PVC) access only to the hosted control plane namespace. Therefore, KubeVirt CSI components cannot access storage from the other namespaces.
```
Comment (lahinson) [inline modules/hcp-virt-storage-security-isolation.adoc:17]:
```suggestion
* Role-based access control (RBAC) of the KubeVirt CSI cluster controller limits the persistent volume claim (PVC) access to only the hosted control plane namespace. Therefore, KubeVirt CSI components cannot access storage from the other namespaces.
```

**Review by lahinson — COMMENTED** [review]
Nice work, @xenolinux! Just a few minor things for your consideration.

File: modules/hcp-virt-storage-security-isolation.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/hcp-virt-storage-security-isolation.adoc]: Checking on infra cluster vs management cluster.

Comment (xenolinux) [reply modules/hcp-virt-storage-security-isolation.adoc]: "management cluster and infrastructure cluster are different terms, but they can technically refer to the same cluster.
Management cluster is the cluster that the hcp control plane pods and hypershift operator run on
Infrastructure cluster is the cluster that the OpenShift Virtualization VMs run on.
The Management cluster is also the Infrastructure cluster when the VMs and HCP pods run on the same baremetal cluster"

Keeping "infrastructure cluster" as per this SME's explanation.

**Pull request merged**